keras_cv/models/object_detection_3d/center_pillar.py

```diff
@@ -261,3 +160,148 @@ def train_step(self, data):
             loss = self.compute_loss(predictions, y)
         self.optimizer.minimize(loss, self.trainable_variables, tape=tape)
         return self.compute_metrics({}, {}, {}, sample_weight={})
+
+    def predict_step(self, data):
+        return self._prediction_decoder(super().predict_step(data))
+
+    @classproperty
+    def presets(cls):
+        """Dictionary of preset names and configurations."""
+        return copy.deepcopy(backbone_presets)
+
+    @classproperty
+    def backbone_presets(cls):
+        """Dictionary of preset names and configurations of compatible
+        backbones."""
+        return copy.deepcopy(backbone_presets)
+
+
+class MultiClassDetectionHead(keras.layers.Layer):
+    """Multi-class object detection head for CenterPillar.
+
+    This head includes a 1x1 convolution layer for each class which is called
+    on the output of the CenterPillar's backbone. The outputs are per-class
+    prediction heatmaps which must be decoded into 3D boxes.
+
+    Args:
+        num_classes: int, the number of box classes to predict.
+        num_head_bin: list of ints, the number of heading bins to use for each
+            respective box class.
+    """
+
+    def __init__(
+        self,
+        num_classes: int,
+        num_head_bin: Sequence[int],
```

Comment:
Should we just give up on typehints? I saw a lot removed during the Keras Core rebase

Reply:
yeah that sgtm.